### ISSUE TITLE
Tuple model failed to execute

### ISSUE THREAD

**Issue by janaSunrise** [shankarpandala/lazypredict#344]
**Describe the bug**

I am trying to train model, and get the results, but I keep getting `tuple model failed to execute.`

**Expected behavior**

It's supposed to be trained, and give me all predictions and info.

**Screenshots**
![Screenshot from 2021-04-09 19-34-20](https://user-images.githubusercontent.com/56873669/114192206-d8558b00-996a-11eb-99c9-c94d75fed9e7.png)
![Screenshot from 2021-04-09 19-34-15](https://user-images.githubusercontent.com/56873669/114192214-da1f4e80-996a-11eb-9df0-1edea524b29e.png)

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser Firefox
 - Version - Latest

**Additional context**

**Comment by janaSunrise** [comment]
I found the cause by doing this
![Screenshot from 2021-04-10 00-07-37](https://user-images.githubusercontent.com/56873669/114226178-e1595300-9990-11eb-8b7c-083ee7b0f2f6.png)
![Screenshot from 2021-04-10 00-07-31](https://user-images.githubusercontent.com/56873669/114226180-e28a8000-9990-11eb-99a6-a6d3a7287547.png)

Hope it helps!

**Comment by Deepankar-98** [comment]
Hi @janaSunrise ,

I found the root cause of this issue and how to fix it.
![image](https://user-images.githubusercontent.com/63971037/132628159-aa3dc856-79d8-433d-9d03-fe891a84fc96.png)

According to the image attached by you, you have used the fit function twice and the 1st red arrow signifies the wrong function call as X_train is used twice instead of X_test.

Replace the 2 red arrows with this line:

```python
models_train, predictions_train = reg.fit(X_train, X_test, y_train, y_test)
```
I was also getting the same error previously because of the same mistake. On making these changes it's working fine for me.
![image](https://user-images.githubusercontent.com/63971037/132628586-998a117e-1d21-4177-bbf1-05b2becbb7d7.png)

Hope it solves your issue also...

**Comment by janaSunrise** [comment]
Hi there, I have both lines, because I want to predict on both Train and Test set, Because I have these lines,
```py
st.subheader("Performance achieved by the model")

    st.write("Training section")
    st.write(predictions_train)
    st.markdown(
        download_file(predictions_train, "training.csv"), unsafe_allow_html=True
    )

    st.write("Testing")
    st.write(predictions_test)
    st.markdown(download_file(predictions_test, "test.csv"), unsafe_allow_html=True)
```
So, Instantiating 2 models helped instead!

**Comment by Deepankar-98** [comment]
Yes in case you want both then creating 2 models will be better.
